Rock/Web/UI/Controls/SearchField.cs

```diff
@@ -58,13 +60,25 @@ public override void RenderControl( HtmlTextWriter writer )
             hfFilter.ID = this.ID + "_hSearchFilter";
 
             var searchExtensions = new Dictionary<string,Tuple<string, string>>();
-            foreach ( KeyValuePair<int, Lazy<Rock.Search.SearchComponent, Rock.Extension.IComponentData>> service in Rock.Search.SearchContainer.Instance.Components )
+
+            var rockPage = this.Page as RockPage;
+            if ( rockPage != null )
             {
-                if ( !service.Value.Value.AttributeValues.ContainsKey( "Active" ) || bool.Parse( service.Value.Value.AttributeValues["Active"].Value ) )
+                var currentPerson = rockPage.CurrentPerson;
+                if ( currentPerson != null )
                 {
-                    searchExtensions.Add( service.Key.ToString(), Tuple.Create<string, string>( service.Value.Value.SearchLabel, service.Value.Value.ResultUrl ) );
-                    if ( string.IsNullOrWhiteSpace( hfFilter.Value ) )
-                        hfFilter.Value = service.Key.ToString();
+                    foreach ( KeyValuePair<int, Lazy<Rock.Search.SearchComponent, Rock.Extension.IComponentData>> service in Rock.Search.SearchContainer.Instance.Components )
+                    {
+                        if ( service.Value.Value.IsAuthorized( Authorization.VIEW, currentPerson ) )
+                        {
```

Comment:
Might want to consider creating\setting a local variable equal to `service.Value.Value` and use it instead since it is being used so often..
`var searchComponent = service.Value.Value;`